File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.py
from typing import *


class Solution:
    def minimumSumSubarray(self, nums: List[int], l: int, r: int) -> int:
        n = len(nums)
        ans = -1
        for i in range(n):
            length = 0
            s = 0
            for j in range(i, n):
                length += 1
                s += nums[j]
                if l <= length <= r and s > 0:
                    if ans == -1 or ans > s:
                        ans = s
        return ans

File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.java
class Solution {
    public int minimumSumSubarray(List<Integer> nums, int l, int r) {
        int n = nums.size();
        int min = Integer.MAX_VALUE;
        for(int i = 0;i < n;i++){
            int s = 0;
            for(int j = i;j < n;j++){
                s += nums.get(j);
                if(s > 0 && j-i+1 >= l && j-i+1 <= r){
                    min = Math.min(min, s);
                }
            }
        }
        if(min == Integer.MAX_VALUE)return -1;
        return min;
    }
}